ansible/rancher/rancher-playbook.yml

```diff
@@ -0,0 +1,101 @@
+---
+- hosts: localhost
+  connection: local
+  gather_facts: false
+  vars_files:
+    - vars.yaml # Assuming this file contains rancher_version and cert_manager_version
+  vars:
+    # Get the current Terraform workspace name
+    terraform_workspace: "{{ lookup('env', 'TF_WORKSPACE') }}"
+    # Construct the path to the workspace's state file
+    terraform_state_file: "../../terraform/aws/cluster_nodes/terraform.tfstate.d/{{ terraform_workspace }}/terraform.tfstate"
+    fqdn: "{{ lookup('cloud.terraform.tf_output', 'fqdn', state_file=terraform_state_file) }}"
+  tasks:
+    - name: Check if kubeconfig file exists
+      stat:
+        path: "{{ kubeconfig_file }}"
+      register: kubeconfig_file_stat
+
+    - name: Fail if kubeconfig file does not exist
+      fail:
+        msg: "kubeconfig file '{{ kubeconfig_file }}' does not exist."
+      when: not kubeconfig_file_stat.stat.exists
+
+    - name: Install cert-manager CRDs
+      kubernetes.core.k8s:
+        kubeconfig: "{{ kubeconfig_file }}"
+        src: https://github.com/cert-manager/cert-manager/releases/download/v{{ cert_manager_version }}/cert-manager.crds.yaml
+        state: present
+      when: cert_manager_version != ""
+
+    - name: Install cert-manager
+      kubernetes.core.k8s:
+        kubeconfig: "{{ kubeconfig_file }}"
+        src: https://github.com/cert-manager/cert-manager/releases/download/v{{ cert_manager_version }}/cert-manager.yaml
+        state: present
+        wait: true
+        wait_timeout: 300
+      when: cert_manager_version != ""
+
+    - name: Wait for cert-manager to be ready
+      kubernetes.core.k8s_info:
+        kubeconfig: "{{ kubeconfig_file }}"
+        api_version: apps/v1
+        kind: Deployment
+        namespace: cert-manager
+        name: cert-manager
+      register: cert_manager_deployment
+      until: cert_manager_deployment.resources[0].status.readyReplicas == cert_manager_deployment.resources[0].status.replicas
+      retries: 10
+      delay: 30
+      when: cert_manager_version != ""
+
+    - name: Add Rancher Helm repository
+      kubernetes.core.helm_repository:
+        kubeconfig: "{{ kubeconfig_file }}"
+        name: "{{ rancher_chart_repo }}"
+        repo_url: "{{ rancher_chart_repo_url }}"
+      when: rancher_version != ""
+
+    - name: Install Rancher
+      kubernetes.core.helm:
+        kubeconfig: "{{ kubeconfig_file }}"
+        name: rancher
+        chart_ref: "{{ rancher_chart_repo }}/rancher"
+        release_namespace: cattle-system
+        create_namespace: true
+        chart_version: "{{ rancher_version }}"
+        wait: true
+        wait_timeout: "600s"
+        values:
+          hostname: "{{ fqdn }}"
+          bootstrapPassword: "{{ bootstrap_password }}"
+      when: rancher_version != ""
```

Comment:
Seems like this `rancher_version` is used both for helm repo configuration and helm installation, same for `cert_manager_version` or other conditions:

If we had a host that already had this repo configuration set, would it still run that step?

Is there a way to enhance these `when` conditions, something for the future - especially if we decide to have more granular playbooks, f.e, airgap and rancher installation playbooks that are calling a helm-configure-and-install playbook

Reply:
Even if we had a repo set we need the version to what we're installing. Even for air gap we take a version for either. I guess I'm not understanding what the concern is.

Reply:
So what I am asking is, why does it have to use the same condition both for configuring and installing the helm chart? These two are separate tasks, but their condition is the same

- If the variable is not empty and the helm repo is not configured -> configure the helm repo
- If the variable is not empty and helm repo is not already installed on the cluster -> install the helm chart

Are these even possible to configure?